### PR TITLE
EUI-3796 WA pact tests added 

### PR DESCRIPTION
**Pull request by sreekanthpuligadda**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

https://tools.hmcts.net/jira/browse/EUI-3796 

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/test/pact/pact-tests/wa/assignTask.spec.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    const taskId = "12345678";
+    describe("post /task/taskId/assign", () => {
+
+        const jwt = 'some-access-token';
+
+        before(async () => {
+            await pactSetUp.provider.setup()
+            const interaction = {
+                state: "some tasks exists on event for case",
```
Comment (BenNzewi) [inline api/test/pact/pact-tests/wa/assignTask.spec.ts:23]: the provider state for this is "assign a task using taskId"

File: api/test/pact/pact-tests/wa/cancelTask.spec.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    const taskId = "12341234"
+    describe("post /task/taskId/cancel", () => {
+
+        const jwt = 'some-access-token';
+
+        before(async () => {
+            await pactSetUp.provider.setup()
+            const interaction = {
+                state: "some tasks exists on event for case",
```
Comment (BenNzewi) [inline api/test/pact/pact-tests/wa/cancelTask.spec.ts:23]: the provider state for this is "cancel a task using taskId"

File: api/test/pact/pact-tests/wa/claimTask.spec.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    const taskId = "121234345656"
+    describe("post /task/taskId/claim", () => {
+
+        const jwt = 'some-access-token';
+
+        before(async () => {
+            await pactSetUp.provider.setup()
+            const interaction = {
+                state: "some tasks exists on event for case",
```
Comment (BenNzewi) [inline api/test/pact/pact-tests/wa/claimTask.spec.ts:23]: the provider state for this is "claim a task using taskId"

File: api/test/pact/pact-tests/wa/completeTask.spec.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    const taskId = "1212343412123434"
+    describe("post /task/taskId/complete/", () => {
+
+        const jwt = 'some-access-token';
+
+        before(async () => {
+            await pactSetUp.provider.setup()
+            const interaction = {
+                state: "some tasks exists on event for case",
```
Comment (BenNzewi) [inline api/test/pact/pact-tests/wa/completeTask.spec.ts:23]: the provider state for this is "complete a task using taskId"

File: api/test/pact/pact-tests/wa/getTaskById.spec.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    describe("get /task/{taskId}", () => {
+
+        const jwt = 'some-access-token';
+
+        before(async () => {
+            await pactSetUp.provider.setup()
+            const interaction = {
+                state: "a valid task exists",
```
Comment (BenNzewi) [inline api/test/pact/pact-tests/wa/getTaskById.spec.ts:45]: the provider state for this is "appropriate task is returned"

File: api/test/pact/pact-tests/wa/assignTask.spec.ts (hunk 1 above)
Comment (sreekanthpuligadda) [reply api/test/pact/pact-tests/wa/assignTask.spec.ts:23]: updated

File: api/test/pact/pact-tests/wa/searchForCompletableTasks.spec.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    describe("post /task/search-for-completable", () => {
+
+        const jwt = 'some-access-token';
+
+        before(async () => {
+            await pactSetUp.provider.setup()
+            const interaction = {
+                state: "some tasks exists on event for case",
```
Comment (BenNzewi) [inline api/test/pact/pact-tests/wa/searchForCompletableTasks.spec.ts:54]: the provider state for this is "appropriate tasks are returned by search for completable"

File: api/test/pact/pact-tests/wa/searchTasks.spec.ts [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    describe("post /task", () => {
+
+        const jwt = 'some-access-token';
+
+        before(async () => {
+            await pactSetUp.provider.setup()
+            const interaction = {
+                state: "some tasks exists",
```
Comment (BenNzewi) [inline api/test/pact/pact-tests/wa/searchTasks.spec.ts:65]: the provider state for this is "appropriate tasks are returned by criteria"

File: api/test/pact/pact-tests/wa/unclaimTask.spec.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    const taskId = "1111222233334444"
+    describe("post /task/taskId/unclaim", () => {
+
+        const jwt = 'some-access-token';
+
+        before(async () => {
+            await pactSetUp.provider.setup()
+            const interaction = {
+                state: "some tasks exists on event for case",
```
Comment (BenNzewi) [inline api/test/pact/pact-tests/wa/unclaimTask.spec.ts:23]: the provider state for this is "unclaim a task using taskId"

File: config/default.json [hunk 9]
```diff
@@ -187,7 +187,7 @@
   ],
   "pact": {
     "branchName": "Dev",
```
Comment (satyachundur) [inline config/default.json:189]: This branch name needs to be Master and in the pact publish you need to check logic publish only if its master

Comment (sreekanthpuligadda) [reply config/default.json:189]: updated branch name to master

Comment (sreekanthpuligadda) [reply config/default.json:189]: added logic in publish.ts to publish only if branchName/pactTag returned is "master"

**Review by udaydenduluri33 — APPROVED** [review]
LGTMs
I can see the PACT tests now deployed

**Review by leepavlouhmcts — DISMISSED** [review]

File: api/test/pact/pact-tests/wa/cancelTask.spec.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            response.then((axiosResponse) => {
+                expect(axiosResponse.status).to.be.equal(204);
+
+            }).then(() => {
+                pactSetUp.provider.verify()
+                pactSetUp.provider.finalize()
+            })
+        })
+    })
+})
```
Comment (connorpgpmcelroy) [inline api/test/pact/pact-tests/wa/cancelTask.spec.ts:59]: Needs newline?

File: api/test/pact/pact-tests/wa/completeTask.spec.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+            response.then((axiosResponse) => {
+                expect(axiosResponse.status).to.be.equal(204);
+                
+            }).then(() => {
+                pactSetUp.provider.verify()
+                pactSetUp.provider.finalize()
+            })
+        })
+    })
+})
```
Comment (connorpgpmcelroy) [inline api/test/pact/pact-tests/wa/completeTask.spec.ts:59]: Newline?

File: api/test/pact/pactUtil.ts [hunk 12]
```diff
@@ -69,3 +69,62 @@ export async function getIdamUsersByEmail(taskUrl: string) {
...
+      'Authorization': 'Bearer some-access-token',
+      'Content-Type': 'application/json',
+      'ServiceAuthorization': 'some service authorisation',
+    },
+  }
+
+  let response: AxiosResponse
+  response = await axios.post(taskUrl, payLoad, axiosConfig)
+  return response
+}
```
Comment (connorpgpmcelroy) [inline api/test/pact/pactUtil.ts:130]: Newline?

**Review by connorpgpmcelroy — COMMENTED** [review]
Newlines needed? Unsure whether that has to do with linting not being run as mentioned before?

**Review by Vamshi-Muniganti — DISMISSED** [review]
reviewed the tests the changes and tests looks ok to me

**Review by leepavlouhmcts — DISMISSED** [review]
approved

**Review by udaydenduluri33 — APPROVED** [review]
LGTMe

**Review by udaydenduluri33 — APPROVED** [review]
LGTMs

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

**Review by Vamshi-Muniganti — APPROVED** [review]
looks good to me

**Pull request merged**